easybuild/easyconfigs/p/poetry/poetry-1.5.1-GCCcore-12.3.0.eb

```
@@ -0,0 +1,187 @@
+easyblock = 'PythonBundle'
+
+name = 'poetry'
+version = '1.5.1'
+
+homepage = 'https://python-poetry.org'
+description = """Python packaging and dependency management made easy. Poetry helps you declare, manage and install
+ dependencies of Python projects, ensuring you have the right stack everywhere."""
+
+toolchain = {'name': 'GCCcore', 'version': '12.3.0'}
+
+builddependencies = [
+    ('binutils', '2.40'),
+    ('pkgconf', '1.9.5'),
+    ('scikit-build', '0.17.6'),
+]
+
+dependencies = [
+    ('Python', '3.11.3'),
+    ('hatchling', '1.18.0'),
+    ('OpenSSL', '1.1', '', SYSTEM),
+    ('cryptography', '41.0.1'),
+]
+
+use_pip = True
+sanity_pip_check = True
+
+exts_list = [
+    ('six', '1.16.0', {
+        'checksums': ['1e61c37477a1626458e36f7b1d82aa5c9b094fa4802892072e49de9c60c4c926'],
+    }),
+    ('idna', '3.4', {
+        'checksums': ['814f528e8dead7d329833b91c5faa87d60bf71824cd12a7530b5526063d02cb4'],
+    }),
+    ('certifi', '2023.5.7', {
+        'checksums': ['0f0d56dc5a6ad56fd4ba36484d6cc34451e1c6548c61daad8c320169f91eddc7'],
+    }),
+    # poetry 1.5.1 has requirement urllib3<2.0.0,>=1.26.0
+    ('urllib3', '1.26.16', {
+        'checksums': ['8f135f6502756bde6b2a9b28989df5fbe87c9970cecaa69041edcce7f0589b14'],
+    }),
+    ('charset-normalizer', '3.1.0', {
+        'checksums': ['34e0a2f9c370eb95597aae63bf85eb5e96826d81e3dcf88b8886012906f509b5'],
+    }),
+    ('flit_scm', '1.7.0', {
+        'checksums': ['961bd6fb24f31bba75333c234145fff88e6de0a90fc0f7e5e7c79deca69f6bb2'],
+    }),
+    ('dulwich', '0.21.5', {
+        'checksums': ['70955e4e249ddda6e34a4636b90f74e931e558f993b17c52570fa6144b993103'],
+    }),
+    ('docutils', '0.20.1', {
+        'checksums': ['f08a4e276c3a1583a86dce3e34aba3fe04d02bba2dd51ed16106244e8a923e3b'],
+    }),
+    ('tomli_w', '1.0.0', {
+        'checksums': ['f463434305e0336248cac9c2dc8076b707d8a12d019dd349f5c1e382dd1ae1b9'],
+    }),
+    ('flit', '3.9.0', {
+        'checksums': ['d75edf5eb324da20d53570a6a6f87f51e606eee8384925cd66a90611140844c7'],
+    }),
+    ('crashtest', '0.4.1', {
+        'source_tmpl': SOURCE_PY3_WHL,
+        'checksums': ['8d23eac5fa660409f57472e3851dab7ac18aba459a8d19cbbba86d3d5aecd2a5'],
+    }),
+    ('zipp', '3.15.0', {
+        'checksums': ['112929ad649da941c23de50f356a2b5570c954b65150642bccdd66bf194d224b'],
+    }),
+    ('importlib_metadata', '6.7.0', {
+        'checksums': ['1aaf550d4f73e5d6783e7acb77aec43d49da8017410afae93822cc9cca98c4d4'],
+    }),
+    ('jeepney', '0.8.0', {
+        'source_tmpl': SOURCE_PY3_WHL,
+        'checksums': ['c0a454ad016ca575060802ee4d590dd912e35c122fa04e70306de3d076cce755'],
+    }),
+    ('semantic_version', '2.10.0', {
+        'checksums': ['bdabb6d336998cbb378d4b9db3a4b56a1e3235701dc05ea2690d9a997ed5041c'],
+    }),
+    ('pyasn1', '0.5.0', {
+        'checksums': ['97b7290ca68e62a832558ec3976f15cbf911bf5d7c7039d8b861c2a0ece69fde'],
+    }),
```

In my tests, these extensions can be removed from `poetry`. They do not seem to be a requirement.
```suggestion
    ('dulwich', '0.21.5', {
        'checksums': ['70955e4e249ddda6e34a4636b90f74e931e558f993b17c52570fa6144b993103'],
    }),
    ('crashtest', '0.4.1', {
        'source_tmpl': SOURCE_PY3_WHL,
        'checksums': ['8d23eac5fa660409f57472e3851dab7ac18aba459a8d19cbbba86d3d5aecd2a5'],
    }),
    ('zipp', '3.15.0', {
        'checksums': ['112929ad649da941c23de50f356a2b5570c954b65150642bccdd66bf194d224b'],
    }),
    ('importlib_metadata', '6.7.0', {
        'checksums': ['1aaf550d4f73e5d6783e7acb77aec43d49da8017410afae93822cc9cca98c4d4'],
    }),
    ('jeepney', '0.8.0', {
        'source_tmpl': SOURCE_PY3_WHL,
        'checksums': ['c0a454ad016ca575060802ee4d590dd912e35c122fa04e70306de3d076cce755'],
    }),
```

---

Yes, we can remove these extensions for `poetry`. But for building `Python-bundle-PyPI` we need `flit_scm` and `flit`. I think the best option is to add another easyconfig for `flit` and add this as build dependency to `Python-bundle-PyPI`.